### PR TITLE
[WIP] Planet turf reworks / additions

### PR DESCRIPTION
**Pull request by Citinited**
TO DO:
- Find some way of improving load times (Currently ~0.4 seconds, I feel it can be improved)
- Make borders between different turfs on the same level smooth nicely

This PR does a bunch of (mostly behind-the-scenes) stuff related to the terrain! WIP as I think I'll try to improve how much time it currently takes to initialize this (~0.4 seconds on an i5-6500, for reference). Let me know if you have any thoughts!

- Grass and dirt tiles now have their own overlay border, for fancy paths and
such;
- Adds indestructible bedrock, and sets it as the baseturf;
- Adds LOW_TURF_LAYER for bedrock;
- Planet turfs now have their layer var set according to their relative elevation, and apply their overlay accordingly;
- Reworks a bunch of planet turf code to be more unified;
- Explosions on the planet now look a lot cooler;
- Adds a lot of new turf types for use in mapping.

Map changes:
- Everything now has its own specific turf instead of being map-edited versions of the base `turf/open/floor/planet` turf. 💢 

Pictures!

Explosions, in order of size:
http://i.imgur.com/lDeBTHf.png
http://i.imgur.com/TcsTeEI.png
http://i.imgur.com/4IfQbTy.png
http://i.imgur.com/c5CyIeY.png

Processing time needed with:

And without:

:cl:
tweak: The terrain looks much nicer now.
add: Adds a bunch of terrain types.
/:cl:

### CODE REVIEW

File: code/game/turfs/simulated/planet.dm [hunk 1]
```diff
@@ -22,81 +26,101 @@
...
 		return
-	else if(istype(C, /obj/item/weapon/shovel) && params)
-		return
+	else if(istype(C, /obj/item/weapon/shovel))
+		if(!candigturf)
+			to_chat(user, "<span class = 'warning'>You can't dig that!</span>")
+			return ..()
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, 1)
+		if(do_after(user, 20, target = src) && candigturf)
+			user.visible_message("<span class = 'notice'>[user] digs up [src].</span>", "<span class = 'notice'>You [src.turfverb] [src].</span>")
```
Comment (SamHPurp) [inline code/game/turfs/simulated/planet.dm]: Might be being thick but if you're assigning target as src in the previous line, shouldn't you just `[target]` and `[target.turfverb]`

File: code/game/turfs/simulated/planet.dm [hunk 2]
```diff
@@ -22,81 +26,101 @@
...
+			digDown(dropsloot = TRUE) //Make some sand after the user digs up some turf
```
Comment (SamHPurp) [inline code/game/turfs/simulated/planet.dm]: It'd be good to get soil or something rather than sand, but that obviously doesn't exist yet ;)

Comment (Citinited) [inline code/game/turfs/simulated/planet.dm]: Mm. Soil would be good for filling in holes and stuff like that.

File: code/game/turfs/simulated/planet.dm (hunk 1 above)
Comment (Citinited) [inline code/game/turfs/simulated/planet.dm]: It's pre-existing code so I didn't really look at it in much detail, but I'll see about improving it

Comment (SamHPurp) [inline code/game/turfs/simulated/planet.dm]: Cool, take a look... when you're satisfied ping me

Comment (Citinited) [inline code/game/turfs/simulated/planet.dm]: For reference, the `target = src` has nothing to do with the actual digging of the turf; it is merely for making the loading bar popup thingy with `do_after`.

**Comment by FlattestGuitar** [comment]
LGTM so far.

**Pull request merged**